src/custom_modules/DAO_modules/distributions.ts

```diff
@@ -617,6 +568,76 @@ async function add(
   }
 }
 //endregion
+export type DistributionDbResultRow = Distributions &
+  Omit<Distribution_cause_areas, "Percentage_share"> &
+  Omit<Distribution_cause_area_organizations, "Percentage_share"> & {
+    Cause_area_percentage_share: Prisma.Decimal;
+    Organization_percentage_share: Prisma.Decimal;
+  };
+export type DistributionDbResult = DistributionDbResultRow[];
+
+const mapDbDistributionsToDistributions = (
+  result: SqlResult<DistributionDbResult>,
+): Distribution[] => {
+  /**
+   * First we map the result to a map of KID -> DistributionDbResult
+   * Such that we have a map of all the rows for a given KID
+   */
+  const map = new Map<string, SqlResult<DistributionDbResult>>();
+
+  result.forEach((row) => {
+    if (!map.has(row.KID)) {
+      map.set(row.KID, []);
+    }
+
+    map.get(row.KID)?.push(row);
+  });
+
+  /**
+   * Then we map each of the rows for a given KID to a Distribution
+   */
+  const distributions: Distribution[] = [];
+  map.forEach((rows) => {
+    distributions.push(mapDbDistributionToDistribution(rows));
+  });
+
+  return distributions;
+};
+
+/**
+ * An important assumption here is that all the rows in the result have the same KID
+ * If you have multiple distributions returned from DB, use mapDbDistributionsToDistributions
+ */
+const mapDbDistributionToDistribution = (result: SqlResult<DistributionDbResult>): Distribution => {
```

Comment:
Is there any chance an empty array is passed as argument? Perhaps we should validate array length and throw an explicit error or return null.